nntrainer/include/optimizer.h

```diff
@@ -161,8 +161,8 @@ class Optimizer {
    * @param[in/out] Bias Bias Tensor
    * @param[in] iteration nth epoch number
    */
-  void calculate(const Tensor &djdw, const Tensor &djdb, Tensor &weight,
-                 Tensor &bias, int iteration);
+  void calculate_gradients(std::vector<std::pair<Tensor, Tensor>> val_grad,
```

Comment:
Would this be applicable to avoid memcopy and update the given weight?
```suggestion
  void calculate_gradients(std::vector<std::pair<Tensor & , const Tensor &>> val_grad,
```

Reply:
updated to `std::vector<Tensor> &weights, std::vector<Tensor> &gradients` as separate list of gradients will be available in each layer anyways in future.
Note: `std::vector<Tensor &>` cant be done as the type `Tensor &` wont be assignable.

Reply:
Updated to `std::vector<std::reference_wrapper<Tensor>> &weights, std::vector<std::reference_wrapper<Tensor>> &gradients` as per offline discussion.